opentelemetry-sdk/tests/metrics/export/test_export.py

```diff
@@ -243,13 +254,58 @@ def test_merge(self):
         counter.merge(counter2)
         self.assertEqual(counter.checkpoint, 4.0)
 
+    def test_concurrent_update(self):
+        counter = CounterAggregator()
+
+        with concurrent.futures.ThreadPoolExecutor(max_workers=2) as executor:
+            fut1 = executor.submit(self.call_update, counter)
+            fut2 = executor.submit(self.call_update, counter)
+
+            updapte_total = fut1.result() + fut2.result()
+
+        counter.take_checkpoint()
+        self.assertEqual(updapte_total, counter.checkpoint)
+
+    def test_concurrent_update_and_checkpoint(self):
+        counter = CounterAggregator()
+        checkpoint_total = 0
+
+        with concurrent.futures.ThreadPoolExecutor(max_workers=1) as executor:
+            fut = executor.submit(self.call_update, counter)
+
+            while not fut.done():
+                counter.take_checkpoint()
+                checkpoint_total += counter.checkpoint
+
+        counter.take_checkpoint()
+        checkpoint_total += counter.checkpoint
+
+        self.assertEqual(fut.result(), checkpoint_total)
+
 
 class TestMinMaxSumCountAggregator(unittest.TestCase):
+    @classmethod
+    def call_update(cls, mmsc):
+        min_ = 2 ** 32
```

Comment:
You might use `float('inf')` and `-float('inf')`.

Reply:
Right, updated.